allennlp/modules/transformer/transformer_stack.py

```diff
@@ -129,67 +134,15 @@ def forward(
         )
 
     @classmethod
-    def _get_input_arguments(
-        cls,
-        pretrained_module: torch.nn.Module,
-        source="huggingface",
-        mapping: Optional[Dict[str, str]] = None,
-        **kwargs,
-    ):
-        submodules = cls._get_mapped_submodules(pretrained_module, source, mapping)
-
+    def _from_config(cls, config: "PretrainedConfig", **kwargs):
         final_kwargs = {}
-
-        final_kwargs["num_hidden_layers"] = len(submodules["layers"])
-
-        final_kwargs["hidden_size"] = submodules["layers.0.attention.self.query"].in_features
-        final_kwargs["num_attention_heads"] = submodules[
-            "layers.0.attention.self"
-        ].num_attention_heads
-        final_kwargs["attention_dropout"] = submodules["layers.0.attention.self.dropout"].p
-        final_kwargs["hidden_dropout"] = submodules["layers.0.attention.output.dropout"].p
-        final_kwargs["intermediate_size"] = submodules["layers.0.intermediate.dense"].out_features
-
-        # We require the if block as `act_fn` is a function rather than a module,
-        # so `_get_mapped_submodules` does not automatically fix this.
-        if source == "huggingface":
-            final_kwargs["activation"] = getattr(
-                submodules["layers.0.intermediate"], "intermediate_act_fn"
-            )
-        else:
-            final_kwargs["activation"] = getattr(submodules["layers.0.intermediate"], "act_fn")
-
-        final_kwargs["add_cross_attention"] = "layers.0.cross_attention" in submodules
-
+        final_kwargs["num_hidden_layers"] = config.num_hidden_layers
+        final_kwargs["hidden_size"] = config.hidden_size
+        final_kwargs["num_attention_heads"] = config.num_attention_heads
+        final_kwargs["add_cross_attention"] = config.add_cross_attention
+        final_kwargs["attention_dropout"] = config.attention_probs_dropout_prob
+        final_kwargs["hidden_dropout"] = config.hidden_dropout_prob
+        final_kwargs["intermediate_size"] = config.intermediate_size
+        final_kwargs["activation"] = config.hidden_act
         final_kwargs.update(**kwargs)
-
-        return final_kwargs
-
-    @classmethod
-    def from_pretrained_module(  # type: ignore
-        cls,
-        pretrained_module: Union[str, torch.nn.Module],
-        num_hidden_layers: Optional[Union[int, range]] = None,
```

Comment:
I think we are losing the ability to load only some of the layers here. I thought we had a test for this; looks like we don't. But this is the functionality I'm talking about: https://github.com/allenai/allennlp/blob/a30dd77584ea3a1866c6499b0acd297e75d8de9d/allennlp/modules/transformer/__init__.py#L72

Reply:
Ah, sorry I should have left a comment about this. This is still possible, and we have a test for it here:

https://github.com/allenai/allennlp/blob/d2f43c71ee89cc6ce2a9749307f75d118f26463d/tests/modules/transformer/toolkit_test.py#L85-L97

I changed this up a bit so that we didn't have to override the `from_pretrained_module` method and change the signature. Also I don't think passing a `range` type would work through config files anyway, right? But if this way is too clunky we could up with alternate, easier way to do this.

Reply:
Ok, this works!